src/dev/create-extension.ts

```diff
@@ -0,0 +1,141 @@
+import arg from "arg";
+import path from "path";
+import fs from "fs";
+import { promisify } from "util";
+import { execa } from "execa";
+import ncp from "ncp";
+import { fileURLToPath } from "url";
+import { SOLIDITY_FRAMEWORKS } from "../utils/consts";
+import chalk from "chalk";
+
+const EXTERNAL_EXTENSIONS_DIR = "externalExtensions";
+const TARGET_EXTENSION_DIR = "extension";
+const TEMPLATE_FILE_SUFFIX = ".template.mjs";
+const BASE_PATH = "base";
+const SOLIDITY_FRAMEWORKS_PATH = "solidity-frameworks";
+const DEPLOYED_CONTRACTS_FILE = "deployedContracts.ts";
+const YARN_LOCK_FILE = "yarn.lock";
+
+const prettyLog = {
+  info: (message: string, indent = 0) => console.log(chalk.cyan(`${"  ".repeat(indent)}${message}`)),
+  success: (message: string, indent = 0) => console.log(chalk.green(`${"  ".repeat(indent)}✔ ${message}`)),
+  warning: (message: string, indent = 0) => console.log(chalk.yellow(`${"  ".repeat(indent)}⚠ ${message}`)),
+  error: (message: string, indent = 0) => console.log(chalk.red(`${"  ".repeat(indent)}✖ ${message}`)),
+};
+
+const ncpPromise = promisify(ncp);
+
+const currentFileUrl = import.meta.url;
+const templateDirectory = path.resolve(decodeURI(fileURLToPath(currentFileUrl)), "../../templates");
+
+const parseArguments = (rawArgs: string[]) => {
+  const args = arg({}, { argv: rawArgs.slice(2) });
+  const projectPath = args._[0];
+  if (!projectPath) {
+    throw new Error("Project path is required");
+  }
+  return { projectPath };
+};
+
+const getChangedFilesFromFirstCommit = async (projectPath: string): Promise<string[]> => {
+  const { stdout: firstCommit } = await execa("git", ["rev-list", "--max-parents=0", "HEAD"], {
+    cwd: projectPath,
+  });
+  const { stdout } = await execa("git", ["diff", "--name-only", `${firstCommit.trim()}..HEAD`], {
+    cwd: projectPath,
+  });
+  return stdout.split("\n").filter(Boolean);
+};
+
+const getChangedFiles = async (projectPath: string): Promise<string[]> => {
```

Comment:
it seems we don't need this function since it only calls getChangedFilesFromFirstCommit

Reply:
Ohh yeah missed it while refactoring, updated at https://github.com/scaffold-eth/create-eth/pull/73/commits/10a808fdf77024e54a68591fc90b32f4cf8260a8